master/internal/usergroup/api_groups.go

```diff
@@ -30,6 +32,15 @@ func (a *UserGroupAPIServer) CreateGroup(ctx context.Context, req *apiv1.CreateG
 		err = apiutils.MapAndFilterErrors(err)
 	}()
 
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = AuthZProvider.Get().CanUpdateGroups(*curUser)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
here (and adding / removing users in ```UpdateGroup```) we technically need a check that a user can view all ```req.AddUsers``` but since all implementations allow viewing all users I think it is fine to leave it out

Reply:
yeah we don't even have a view permission to check on yet - tbh I'm not even sure we need one, but it seems plausible to me you'd require it in the future